src/node.rs

```diff
@@ -168,26 +232,14 @@ impl NodeName {
     }
 }
 
-impl fmt::Display for NodeName {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(
-            f,
-            "{}",
-            match self {
-                NodeName::Path(expr) => path_to_string(expr),
-                NodeName::Dash(name) => name
-                    .iter()
-                    .map(|ident| ident.to_string())
-                    .collect::<Vec<String>>()
-                    .join("-"),
-                NodeName::Colon(name) => name
-                    .iter()
-                    .map(|ident| ident.to_string())
-                    .collect::<Vec<String>>()
-                    .join(":"),
-                NodeName::Block(_) => String::from("{}"),
-            }
-        )
+impl TryFrom<&NodeName> for ExprBlock {
+    type Error = eyre::Report;
```

Comment:
This needs to be converted to an internal error type using `thiserror` and maybe `error-stack`. Which should also help with https://github.com/stoically/syn-rsx/issues/12.